File: sdks/python/playwright/tests/support/CSS.py
import pytest
from applitools.common import StitchMode
from applitools.playwright import Region, Target, ClassicRunner
from playwright.sync_api import Page


@pytest.fixture(scope="function")
def eyes_runner_class():
    return ClassicRunner()


@pytest.fixture(scope="function")
def stitch_mode():
    return StitchMode.CSS


def window(page: Page, eyes):
    page.goto("https://applitools.github.io/demo/TestPages/FramesTestPage/")
    conf = eyes.get_configuration()
    conf.test_name = "Playwright Window - Classic"
    conf.app_name = "Applitools Support Matrix"
    conf.viewport_size = {"width": 700, "height": 460}
    eyes.set_configuration(conf)
    eyes.open(page)
    eyes.check(Target.window())
    eyes.close(raise_ex=True)


def region(page: Page, eyes):
    page.goto("https://applitools.github.io/demo/TestPages/FramesTestPage/")
    conf = eyes.get_configuration()
    conf.test_name = "Playwright Region - Classic"
    conf.app_name = "Applitools Support Matrix"
    conf.viewport_size = {"width": 700, "height": 460}
    eyes.set_configuration(conf)
    eyes.open(page)
    eyes.check(Target.window().region(Region(50, 70, 90, 110)))
    eyes.close(raise_ex=True)


def frame(page: Page, eyes):
    page.goto("https://applitools.github.io/demo/TestPages/FramesTestPage/")
    conf = eyes.get_configuration()
    conf.test_name = "Playwright Frame - Classic"
    conf.app_name = "Applitools Support Matrix"
    conf.viewport_size = {"width": 700, "height": 460}
    eyes.set_configuration(conf)
    eyes.open(page)
    eyes.check(Target.window().frame("frame1"))
    eyes.close(raise_ex=True)

File: sdks/java/playwright/src/test/java/support/tests/CSS.java
package support.tests;

import com.applitools.eyes.RectangleSize;
import com.applitools.eyes.Region;
import com.applitools.eyes.playwright.fluent.Target;
import com.applitools.eyes.selenium.StitchMode;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;
import support.BaseSetup;

public class CSS extends BaseSetup {

    @BeforeClass
    public void classSetup() {
        UFG = false;
        stitchMode = StitchMode.CSS;
    }

    @Test
    public void window() {
        driver.navigate("https://applitools.github.io/demo/TestPages/FramesTestPage/");
        eyes.open(driver, "Applitools Support Matrix", "Playwright Window - Classic", new RectangleSize(700, 460));
        eyes.check(Target.window());
        eyes.close(true);
    }

    @Test
    public void region() {
        driver.navigate("https://applitools.github.io/demo/TestPages/FramesTestPage/");
        eyes.open(driver, "Applitools Support Matrix", "Playwright Region - Classic", new RectangleSize(700, 460));
        eyes.check(Target.region(new Region(50, 70, 90, 110)));
        eyes.close(true);
    }

    @Test
    public void frame() {
        driver.navigate("https://applitools.github.io/demo/TestPages/FramesTestPage/");
        eyes.open(driver, "Applitools Support Matrix", "Playwright Frame - Classic", new RectangleSize(700, 460));
        eyes.check(Target.frame(driver.locator("[name=\"frame1\"]")));
        eyes.close(true);
    }
}
